lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -182,6 +182,30 @@ def get_cpu_facts(self, collected_facts=None):
             return cpu_facts
 
         cpu_facts['processor'] = []
+        cmd = self.module.get_bin_path('lscpu')
+        # cmd = cmd + " | egrep '^(Socket|Thread|Core|CPU|Model name)'"
+        rc, lscpu_out, err = self.module.run_command(cmd)
+        if lscpu_out:
+            for line in lscpu_out.splitlines():
+                data = line.split(' ')
+                if line.startswith('Socket(s)'):
+                    cpu_facts['processor_count'] = data[-1]
```

Comment:
This needs to be an `int` to be consistent with what we return currently. And if we split on colon, we can get the second list element rather than the last element.
```suggestion
                    cpu_facts['processor_count'] = int(data[1])
```

You can use the `type_debug` filter to check the type of Jinja variables.

Reply:
fixed